python/cudf/cmake/Modules/ProtobufHelpers.cmake

```diff
@@ -0,0 +1,49 @@
+# =============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+include_guard(GLOBAL)
+
+# Compile protobuf files to Python. All arguments are assumed to be .proto files.
+function(codegen_protoc)
+  # Allow user to provide path to protoc executable as an environment variable.
+  if(DEFINED ENV{PROTOC})
+    set(protoc_COMMAND $ENV{PROTOC})
+  else()
+    find_program(protoc_COMMAND protoc REQUIRED)
+  endif()
+
+  foreach(_proto_path IN LISTS ARGV)
+    string(REPLACE "\.proto" "_pb2\.py" pb2_py_path "${_proto_path}")
+    set(pb2_py_path "${CMAKE_CURRENT_SOURCE_DIR}/${pb2_py_path}")
+    execute_process(
+      COMMAND ${protoc_COMMAND} --python_out=. "${_proto_path}"
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} COMMAND_ERROR_IS_FATAL ANY
+    )
+    # Mark entire file to skip formatting.
+    file(READ "${pb2_py_path}" pb2_py)
+    file(
+      WRITE "${pb2_py_path}"
+      [=[
+# flake8: noqa
+# fmt: off
+]=]
+    )
+    file(APPEND "${pb2_py_path}" "${pb2_py}")
+    file(
+      APPEND "${pb2_py_path}"
+      [=[
+# fmt: on
+]=]
+    )
+  endforeach()
+endfunction()
```

Comment:
Minor (more for understanding): does this do partial rebuild tracking appropriately? That is, if I only change a single `.proto` file does only that one get rebuilt?

Reply:
Currently it doesn't matter, because there is only one `.proto` file.
https://github.com/rapidsai/cudf/blob/branch-22.12/python/cudf/cudf/utils/metadata/orc_column_statistics.proto

I'm not sure what would happen if multiple files were present, but it may not be worth worrying about.

Reply:
Yeah this will just rebuild everything naively. We can rethink the strategy if we ever need to scale up, but right now the usage is super limited to a single ORC statistics file so it's not worth overengineering yet.